keras/layers/dense_attention_test.py

```diff
@@ -203,6 +203,30 @@ def test_calculate_scores_multi_dim(self):
     expected = np.array([[[7.64, 12.24, 16.84], [14.24, 22.84, 31.44]]],
                         dtype=np.float32)
     self.assertAllClose(expected, actual)
+ 
+  def test_calculate_scores_multi_dim_concat(self):
+    # Query tensor of shape [1, 2, 4]
+    q = np.array([[[1., 1.1, 1.2, 1.3], [2., 2.1, 2.2, 2.3]]], dtype=np.float32)
+    # Key tensor of shape [1, 3, 4]
+    k = np.array(
+        [[[1.5, 1.6, 1.7, 1.8], [2.5, 2.6, 2.7, 2.8], [3.5, 3.6, 3.7, 3.8]]],
+        dtype=np.float32)
+    attention_layer = dense_attention.Attention(score_mode='concat')
+    attention_layer.build(input_shape=([1, 2, 4], [1, 3, 4]))
+    actual = attention_layer._calculate_scores(query=q, key=k)
```

Comment:
Test is failing: https://source.cloud.google.com/results/invocations/acd8077d-49cf-4ad9-a164-2c37718c3a47/targets/keras%2Fgithub%2Fubuntu%2Fcpu%2Fpresubmit/log

You may need to do `actual = keras.backend.get_value(actual)`

Reply:
Is it because I am not initializing the parameter `attention_v` and only create it if `score_mode= 'concat' ` ? Because `score_mode='dot' ` is passing and the scale parameter is also initialized to None at the start itself.

Reply:
So what do you get when you replace it with `actual = keras.backend.get_value(actual)`?

Reply:
To maintain the uniformity in the tests I havent used this yet. 
The error msg is : 

> Node: 'ReadVariableOp'
Could not find variable attention_v. This could mean that the variable has been deleted. In TF1, it can also mean the variable is uninitialized. Debug info: container=localhost, status error message=Container localhost does not exist. (Could not find resource: localhost/attention_v)
	 [[{{node ReadVariableOp}}]]
	 

So initialized `attention_v` (now `concat_score_weight`) to None just like `scale` is done.

Reply:
The error is unrelated to this. It will likely go away if you use `get_value`.

Reply:
Did the requested changes.